cpp/src_prims/stats/histogram.h

```diff
@@ -331,7 +329,9 @@ __global__ void smemHashHistKernel(int* bins, const DataT* data, IdxT nrows,
     }
     __syncthreads();
     if (needFlush[0]) {
+      __syncthreads();
```

Comment:
Do you need this sync?  You've just done one to get needFlush[0] shared among all threads.  What does the middle one do?  I actually thought it was correct without these here nor needing the ones in flushHashTable.

Reply:
My bad. You're right. This syncthreads is not needed. Removed it. Thanks.